crates/driver/src/infra/observe/mod.rs

```diff
@@ -333,13 +333,14 @@ pub fn mempool_executed(
                 ?err,
                 %mempool,
                 ?settlement,
+                block_no=err.block_no(),
```

Comment:
Why do we need this extra line if `?err` would already print the block number contained in the error?